sleuth/sleuth/enrich_tls.py

```diff
@@ -102,82 +208,41 @@ def tls_seclevel(policy, unknowns, scs, client_key_length, certs):
-                    if client_key_length >= int(each):
-                        kex_seclevel = kex_policy[kex]["client_key_length"][each]
-            elif "default" in kex_policy[kex]:
-                kex_seclevel = kex_policy[kex]["default"]
-
-
-        # Analyze seclevel of certificates based on algorithm and key size seclevel
-        # information given in the policy json
-        #
-        if certs:
-            certs_seclevel = max_seclevel
-            sig_policy = policy_rules["sig_alg"]
-
-            for x in certs:
-                sig_alg = x['signature_algorithm']
-                sig_key_size = x['signature_key_size']
-
-                if sig_alg and sig_alg in sig_policy:
-                    if sig_key_size and "sig_key_size" in sig_policy[sig_alg]:
-                        for each in sig_policy[sig_alg]["sig_key_size"]:
-                            if sig_key_size >= int(each):
-                                tmp_seclevel = sig_policy[sig_alg]["sig_key_size"][each]
-                    elif "seclevel" in sig_policy[sig_alg]:
-                        tmp_seclevel = sig_policy[sig_alg]["seclevel"]
-                else:
-                    tmp_seclevel = UNKNOWN
-
-                if tmp_seclevel and tmp_seclevel < certs_seclevel:
-                    certs_seclevel = tmp_seclevel
-        else:
-            certs_seclevel = UNKNOWN
-
-        # Other seclevel items can be assessed in one step due to the lack of nested
-        # seclevel elements; we create an inventory for the flow here
-        #
-        seclevel_inventory = {
-            "kex": kex_seclevel,
-            "certs": certs_seclevel,
-            "sig": policy_rules["sig"][params['sig']] if params['sig'] in policy_rules["sig"] else UNKNOWN,
-            "enc": policy_rules["enc"][params['enc']] if params['enc'] in policy_rules["enc"] else UNKNOWN,
-            "auth": policy_rules["auth"][params['auth']] if params['auth'] in policy_rules["auth"] else UNKNOWN,
-            "hash": policy_rules["hash"][params['hash']] if params['hash'] in policy_rules["hash"] else UNKNOWN
-        }
+        seclevel_inventory = {}
+        concerns = []
+
+        # loop through the items that need to be evaluated against the policy
+        for attr in params:
+            if attr == "desc":
+                continue
+
+            current_seclevel = UNKNOWN
+
+            if attr in policy.rules:
+                if params[attr] in policy.rules[attr]:
+                    if attr == "kex":
+                        current_seclevel = get_seclevel_with_param(policy, attr, params[attr], client_key_length)
+                    else:
+                        current_seclevel = policy.rules[attr][params[attr]]
+
+            # build a list of items whose seclevel falls below the failure_threshold
+            # and the value that caused the failure
+            if current_seclevel <= policy.failure_threshold:
+                concern_string = attr + ": " + params[attr]
```

Comment:
Could you substitute "-" or something for the ":" because the output is JSON and having an extra colon in the text looks a bit confusing